cargo-near-build/src/cargo_native/compile.rs

```diff
@@ -15,44 +15,32 @@ use super::ArtifactType;
 pub fn run<T>(
     manifest_path: &ManifestPath,
     args: &[&str],
-    mut env: Vec<(&str, &str)>,
+    env: Vec<(&str, &str)>,
     hide_warnings: bool,
     color: ColorPreference,
 ) -> eyre::Result<CompilationArtifact<T>>
 where
     T: ArtifactType,
 {
-    let mut final_env = BTreeMap::new();
-
-    if hide_warnings {
-        env.push(("RUSTFLAGS", "-Awarnings"));
-    }
-
-    for (key, value) in env {
-        match key {
-            "RUSTFLAGS" => {
-                let rustflags: &mut String = final_env
-                    .entry(key)
-                    .or_insert_with(|| std::env::var(key).unwrap_or_default());
-                // helps avoids situation on complete match `RUSTFLAGS="-C link-arg=-s -C link-arg=-s"`
-                if !rustflags.contains(value) {
-                    if !rustflags.is_empty() {
-                        rustflags.push(' ');
-                    }
-                    rustflags.push_str(value);
-                }
-            }
-            _ => {
-                final_env.insert(key, value.to_string());
-            }
+    let final_env = {
+        let mut env: BTreeMap<_, _> = env.into_iter().collect();
+        if hide_warnings {
+            env.insert("RUSTFLAGS", "-Awarnings");
         }
-    }
+        env
+    };
+    // removing env, which may be implicitly passed when bulding from within a build-script
+    // see https://github.com/near/cargo-near/issues/287
+    // CARGO_ENCODED_RUSTFLAGS="-Awarnings" and RUSTFLAGS="-Awarnings" both result
+    // in mysterious failures of `cargo build --target wasm32-unknown-unknown` (**cargo** bug)
```

Comment:
Do we need to create an issue for cargo? Or do we have something to link?

Reply:
Let's add this info into the README explaining why we remove that and why. Maybe, some people rely on that functionality.  I doubt it, but it's better to have that than not have :)

Reply:
I'll create an issue to create an issue to cargo. These have to be perfect to have a chance of someone taking a look at them, and those take a lot of time to file in